Mathlib/Tactic/NthRewrite.lean

```diff
@@ -41,7 +78,45 @@ syntax (name := nthRewriteSeq) "nth_rewrite" (config)? ppSpace num rwRuleSeq (lo
   | _ => throwUnsupportedSyntax
 
 /--
-`nth_rw` is like `nth_rewrite`, but also tries to close the goal by trying `rfl` afterwards.
+`nth_rw` is a variant of `nth_rewrite` that also tries to close the goal by trying `rfl` afterwards.
+`nth_rw n [eq₁, eq₂,..., eqₘ]` will rewrite the `n`ᵗʰ _occurrence_ of each of the `m` equalities
+`eqᵢ`in that order. Occurrences are counted beginning with `1`. For example,
+```lean
+example (h : a = 1) : a + a + a + a + a = 5 := by
+  nth_rw 2 [h]
+/-
+a: ℕ
+h: a = 1
+⊢ a + 1 + a + a + a = 5
+-/
+```
+Notice that the second occurrence of `a` from the left has been rewritten by `nth_rewrite`.
+
+If a term `t` is introduced by rewriting with `eqᵢ`, then this instance of `t` will be counted as an
+_occurrence_ of `t` for all subsequent rewrites of `t` with `eqⱼ` for `j > i`. This behaviour is
+illustrated by the example below
+```lean
+example (h : a = a + b) : a + a + a + a + a = 0 := by
+  nth_rw 3 [h, h]
+/-
+a b: ℕ
+h: a = a + b
+⊢ a + a + (a + b + b) + a + a = 0
+-/
+```
+In this example, the first `nth_rw` with `h` introduces an additional occurrence of `a` in
+the goal. That is, the goal state after the first rewrite looks like below
+```lean
+/-
+a b: ℕ
+h: a = a + b
+⊢ a + a + (a + b) + a + a = 0
+-/
+```
+This new instance of `a` also turns out to be the third _occurrence_ of `a`.  Therefore,
+the next `nth_rw` with `h` rewrites this `a`.
+
+Further, `nth_rw` will close the remaining goal with `rfl` if possible.
```

Comment:
I would suggest to `import Mathlib.Tactic.ExtendDoc` and then use `@[inherit_doc nthRewriteSeq]` on the`nth_rw` macro along with the something like the following just below it:

```
extend_docs nthRwSeq before
  "`nth_rw` is like `nth_rewrite`, but also tries to close the goal by trying `rfl` afterwards.
  "
```
That way you could avoid writing that much documentation twice. The user will first see the sentense about `nth_rw` being essentially `nth_rewrite` followed by the doc for the latter.

Reply:
That sounds nice, but it doesn't adapt the examples very well. I was hoping the documentation could be entirely self-contained.

Reply:
I have to admit, I didn't even read the second docstring. If it is merely a `nth_rewrite --> nth_rw` renaming, I think this suggestion would make a lot of sense and wouldn't be confusing for the reader. And it would simplify maintainability (e.g. I think the last example of the second docstring is already out-of-sync again)

However, if there is more change to it might not be ideal.

Anyways, this is a completely _optional_ suggestion!

Reply:
Even better, just have a single doc-string, copied via `inherit_doc` that explains both together. e.g.

`nth_rw` and `nth_rewrite` are analogues of `rw` and `rewrite` that allowing restrict which occurrence of a patttern to replace. `nth_rw` tries to close the goal after rewriting with a reducible transparency `rfl`, while `nth_rewrite` does not (just as for `rw` and `rewrite`).

.... now continue with examples just in terms of `nth_rw`.

Reply:
The difference is the doc tooltip only shows you the docstring for the tactic your mouse pointer is hovering on. So if I am hovering on nth_rw, I would like to see a full explanation rather than being redirected. If inherit doc could accept parameters and format strings, then the docstrings would be correct, especially the examples. My rule of thumb for documentation examples is that they should be copy pastable with minimum modification.

Reply:
I think now that we have the customized docstrings, the question is whether they will be diverging, so that one is updated and the other isn't. That's potentially more harm than copypastability breaking. My estimate is that we'll be okay with two similar docstrings, so I'm inclined to keep these as-is.

Reply:
Kim  has reacted with a (y) to @Vierkantor 's suggestion to keep these strings as is. Shall I assume that this can be marked resolved?